dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultAnalyticsService.java

```diff
@@ -75,21 +76,30 @@
 
     private final AnalyticsCache analyticsCache;
 
+    private final ExpressionDimensionItemAnalyticsCache expressionDimensionItemAnalyticsCache;
+
     private final DataAggregator dataAggregator;
 
     // -------------------------------------------------------------------------
     // AnalyticsService implementation
     // -------------------------------------------------------------------------
 
     @Override
-    public Grid getAggregatedDataValues( DataQueryParams params )
+    public Grid getAggregatedDataValues( DataQueryParams dataQueryParams )
     {
-        params = checkSecurityConstraints( params );
+        final DataQueryParams params = checkSecurityConstraints( dataQueryParams );
 
         queryValidator.validate( params );
 
         if ( analyticsCache.isEnabled() && !params.analyzeOnly() )
         {
+            // expression dimension items will be cached along their DataQueryParams keys (getKey())
+            // when the expression dimension item content is modified (expression dimension item controller),
```

Comment:
We could specify the exact controller class using {@link classname} to be more precise.